deploy-if-version-bump.sh

```diff
@@ -0,0 +1,13 @@
+set -e
+
+# check if CHANGELOG has been updated in current branch
+git diff --name-only master | grep -s CHANGELOG.md || EXIT_CODE=$?
+
+# check if CHANGELOG has been updated in last merge (where HEAD = last merge)
+# git log -m -1 --name-only --pretty="format:" HEAD | grep -s CHANGELOG.md || EXIT_CODE=$?
+
+if [ -z $EXIT_CODE ]; then
```

Comment:
Guys, for the _exact_ same reason I mentioned last time, I think this is setting ourselves for future trouble.

> With this code, you will publish changes only if ~the version has been bumped~ the Changelog has been edited. It seems like future trouble for the person in 2 months who will wonder why their changes are not deployed in production, while the tests are passing on circle !

You are **not** preventing anyone from merging a branch into master without editing the Changelog. You are _just_ not deploying their change. Until someone else edits the Changelog... I don't see _any_ situation where that strategy is relevant.

Now if you _really_ want to enforce changelog editing at each merge (which I find a little overkill, but why not), you want to add a _test_ on it, like [here](https://github.com/openfisca/openfisca-core/blob/master/check-version-bump.sh#L14). You don't need any condition on _deployment_.

<sup><inf>(For information we needed to add such a condition for Python packages only to avoid false red tests when the deployment was failing trying to publish an already existing version: you will _not_ have this issue with a website).</inf></sup>

Reply:
This was a commit to delete. The title of this PR was updated to [WIP] 😕

Reply:
Then I don't understand why this PR was approved last night around midnight, and why I, being the only person having requested changes, was poked right after. 

I'm sure that this was an effort to try to get things moving forward but, but we should try to refrain ourselves from the late-night urge to merge PR.